AutoRest/Generators/Ruby/Azure.Ruby.Tests/RspecTests/lro_spec.rb

```diff
@@ -8,221 +8,433 @@
+    result = @lros_client.put_non_resource(@sku).value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.id).to eq('100')
+    expect(result.body.name).to eq('sku')
+  end
+
+  it 'should succeed for put async non resource' do
+    result = @lros_client.put_async_non_resource(@sku).value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.id).to eq('100')
+    expect(result.body.name).to eq('sku')
+  end
+
+  it 'should succeed for delete provisioning 202 accepted 200' do
+    result = @lros_client.delete_provisioning202accepted200succeeded().value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Succeeded")
+  end
+
+  it 'should succeed for delete provisioning 202 deleting failed 200' do
+    result = @lros_client.delete_provisioning202deleting_failed200().value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Failed")
+  end
+
+  it 'should succeed for delete provisioning 202 deleting canceled 200' do
+    result = @lros_client.delete_provisioning202deletingcanceled200().value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Canceled")
+  end
+
+  it 'should succeed for delete 204' do
+    result = @lros_client.delete204succeeded().value!
+    expect(result.response.status).to eq(204)
+  end
+
+  it 'should succeed for delete 200 retry 200' do
+    result = @lros_client.delete202retry200().value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should succeed for delete 200 no retry 204' do
+    result = @lros_client.delete202no_retry204().value!
+    expect(result.response.status).to eq(204)
+  end
+
+  it 'should succeed for delete async retry' do
+    result = @lros_client.delete_async_retry_succeeded().value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should succeed for delete async no retry' do
+    result = @lros_client.delete_async_no_retry_succeeded().value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should fail for delete async retry failed' do
+    expect { @lros_client.delete_async_retry_failed().value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+
+  it 'should fail for delete async retry canceled' do
+    expect { @lros_client.delete_async_retrycanceled().value! }.to raise_error(MsRestAzure::AzureOperationError)
+  end
+end
+
+describe 'Long Running Operation with retry' do
+  before(:all) do
+    base_url = ENV['StubServerURI']
+
+    dummyToken = 'dummy12321343423'
+    credentials = MsRest::TokenCredentials.new(dummyToken)
+
+    test_client = AutoRestLongRunningOperationTestService.new(credentials, base_url)
+    test_client.long_running_operation_retry_timeout = 0
+    @lroretrys_client = test_client.lroretrys
+    @product = Product.new
+    @product.location = "West US"
   end
 
   # Retryable errors
   it 'should retry PUT request on 500 responce' do

```

Comment:
fixed 

---
In reply to: [69024176](https://github.com/Azure/autorest/pull/1188#discussion_r69024176) [](ancestors = 69024176)